File: test/Test_TRexMeat.cpp
#include "../include/TRexMeat.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(TRexMeatTest, AllFunctionalityofTRexMeat){
    TRexMeat testTM(4000, "King");
    ASSERT_EQ("King", testTM.getProducer());
    ASSERT_EQ(TREX_MEAT, testTM.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testTM.getProductType());
    ASSERT_EQ(4000, testTM.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/meat/TestTRexMeat.java
package engisubs.gameobject.product.farmproduct.meat;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestTRexMeat{
    private TRexMeat test;
    @Before
    public void initObject(){
        test = new TRexMeat(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.TREX_MEAT, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}